DataAccess/DataObject.py

```diff
@@ -120,3 +125,49 @@ def validate_password(cls, password):
             raise DataException()
 
         return res
+
+    @classmethod
+    def find_user(cls,password):
```

Comment:
you should not find a user by their password; the user (email) is either provided in the body or parameter of the request, or directly decoded from the token.